src/server.js

```diff
@@ -139,35 +140,37 @@ class Server extends EventTarget {
       websockets = networkBridge.websocketsLookup(this.url);
     }
 
-    if (typeof options !== 'object' || arguments.length > 3) {
-      data = Array.prototype.slice.call(arguments, 1, arguments.length);
-      data = data.map(item => normalizeSendData(item));
-    } else {
-      data = normalizeSendData(data);
-    }
-
-    websockets.forEach(socket => {
-      if (Array.isArray(data)) {
+    const nonNormalizedData = typeof options !== 'object' || arguments.length > 3
+      ? Array.prototype.slice.call(arguments, 1, arguments.length)
+      : data;
+    const normalizedData = typeof options !== 'object' || arguments.length > 3
+      ? Array.prototype.slice
+        .call(arguments, 1, arguments.length)
+        .map(item => normalizeSendData(item))
+      : normalizeSendData(data);
+
+    websockets.forEach(socket => (_data => {
+      if (Array.isArray(_data)) {
         socket.dispatchEvent(
           createMessageEvent({
             type: event,
-            data,
+            data: _data,
             origin: this.url,
             target: socket.target
           }),
-          ...data
+          ..._data
         );
       } else {
         socket.dispatchEvent(
           createMessageEvent({
             type: event,
-            data,
+            data: _data,
             origin: this.url,
             target: socket.target
           })
         );
       }
-    });
+    })(socket instanceof SocketIO ? nonNormalizedData : normalizedData));
```

Comment:
Self-Invoking Functions look weird and hard to read. Please create another variable instead before passing to the dispatchEvent

Reply:
I agree. This was based on https://github.com/thoov/mock-socket/pull/246/files#r253630182, but I'll change it to something that's easier to read.